infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServer.java

```diff
@@ -359,10 +362,13 @@ public synchronized void resetServer(CorfuPayloadMsg<SequencerTailsRecoveryMsg>
         final boolean bootstrapWithoutTailsUpdate = msg.getPayload()
                 .getBootstrapWithoutTailsUpdate();
 
-        // If bootstrapEpoch is -1 (startup), the sequencer should not accept
-        // bootstrapWithoutTailsUpdate bootstrap messages.
-        if (bootstrapEpoch == Layout.INVALID_EPOCH && bootstrapWithoutTailsUpdate) {
-            log.warn("Cannot update existing sequencer. Require full bootstrap.");
+        // If bootstrapEpoch is -1 (startup) OR readyEpoch is not the consecutive epoch of the
```

Comment:
I think we missed this because this method's logic is convoluted and the language(description of states) is unclear. Since we are here, let's clean this up. The main points: 

- I think sequencerEpoch is more appropriate than readyEpoch
- bootstrapWithoutTailsUpdate should probably be called epochUpdate
- explicitly encode the states (unbootstrapped | streamTailsUpdate )
- reduce the branching complexity (simplify multiple independent if-clauses)

Reply:
bootstrapWithoutTailsUpdate is a boolean denoting whether it is a full or a delta bootstrap message. epochUpdate doesn't seem the right name to replace this.

Reply:
I need the multiple if-statements for appropriate logging. Combining multiple statements will cause difficulty in debugging in cases of bootstrap message rejection. I haven't added any more than the current code.

Reply:
I have made the change to sequencerEpoch.

Reply:
`bootstrapWithoutTailsUpdate should probably be called epochUpdate` Can you please explain?

Reply:
Is this not just a message that just updates the sequencer's epoch? how is delta bootstrap and bootstrapWithoutTailsUpdate more intuitive than epochUpdate. I get what you are trying to say, but I'm pointing out that the language is really overloaded  :)

Reply:
I see. But all bootstrap messages update the epoch - with full bootstrap messages and the delta ones too. Wont it cause confusion?  How about resetSequencerState?